packages/cli/index.js

```diff
@@ -61,16 +65,61 @@ export async function run(...args) {
   }
 
   try {
-    await $`${binaryPath} ${args}`;
+    await spawnAsync(binaryPath, args, { stdio: 'inherit' });
   } catch (error) {
-    if (error instanceof ProcessOutput && error.exitCode) {
-      // console.error({ stderr: error.stderr, stdout: error.stdout });
-      process.exit(error.exitCode);
+    if (error.code) {
+      process.exit(error.code);
     }
     throw error;
   }
 }
 
 export async function uninstall() {
-  await $`rm -rf ${installDirectory}`;
+  await fs.rm(installDirectory, { recursive: true });
+}
+
+export async function checkInstallation() {
+  try {
+    const stats = await fs.lstat(binaryPath);
+    // if the binary is a symlink, it should link to the system-wide zig
+    if (stats.isSymbolicLink()) {
+      console.log('Skipping zig installation, symlink to system zig exists');
+      process.exit(0);
+    }
+    // an empty ./bin/zig file is used as a placeholder for npm/pnpm/yarn to
+    // create the bin symlink, so if the file already exists and has non-zero
+    // size, and is not a symlink, then zig is already installed locally
+    if (stats.size !== 0) {
+      console.log('Skipping zig installation, binary exists');
+      process.exit(0);
+    }
+  } catch {}
+  try {
+    // remove the node_modules/.bin symlinks from the PATH before checking if a
+    // zig is installed system-wide, to avoid hitting the symlink
+    const pathSep = windows ? ';' : ':';
+    const pathDirs = process.env.PATH.split(pathSep);
+    process.env.PATH = pathDirs
+      .filter(dir => !dir.endsWith(path.join('node_modules', '.bin')))
+      .join(':');
+    // check if there's already an installed zig binary
+    const systemZig = await which('zig');
```

Comment:
From the `which` docs, I think instead of overwriting the `process.env.PATH`, we can use the extra path argument instead, what do you think? https://github.com/npm/node-which#options